lib/archethic/transaction_chain/transaction.ex

```diff
@@ -683,71 +685,85 @@ defmodule Archethic.TransactionChain.Transaction do
           9, 228, 44, 237, 157, 90, 243, 90, 6, 0>>
 
   """
-  @spec serialize(t()) :: bitstring()
-  def serialize(%__MODULE__{
-        version: version,
-        address: address,
-        type: type,
-        data: data,
-        previous_public_key: nil,
-        previous_signature: nil,
-        origin_signature: nil,
-        validation_stamp: nil
-      }) do
+  @spec serialize(t(), serialization_mode()) :: bitstring()
+  def serialize(tx, serialization_mode \\ :compact)
+
+  def serialize(
+        %__MODULE__{
+          version: version,
+          address: address,
+          type: type,
+          data: data,
+          previous_public_key: nil,
+          previous_signature: nil,
+          origin_signature: nil,
+          validation_stamp: nil
+        },
+        serialization_mode
+      ) do
     <<version::32, address::binary, serialize_type(type)::8,
```

Comment:
**Suggestion**
Maybe the serialization mode could be added in the serialization, so whenever we need to deserialize a transaction the deserialization will know which mode to use while the program may not know the serialization mode to use before deserialization

Reply:
It can be an idea, but at the end the serialization/deserialization should be the most efficient and will result everytime as the compact mode (except for signature, but this serialization is not persisted)

Reply:
For now I agree it is not usefull. I was thinking for later, but we could change this later if needed